src/main/java/io/sealights/plugins/sealightsjenkins/integration/plugins/SealightsMavenPluginIntegrator.java

```diff
@@ -125,14 +136,28 @@ private StringBuilder addConfigurationToPluginText(StringBuilder plugin) {
 
     private StringBuilder addMetadataToConfigurationInPluginText(StringBuilder plugin) {
         Map<String, String> metadata = new TreeMap<String, String>(pluginInfo.getMetadata());
-        if (!(metadata == null || metadata.isEmpty())) {
-            plugin.append("<metadata>");
-            Iterator it = metadata.entrySet().iterator();
+        return addMapConfigurationToPluginText(plugin,  METADATA_KEY, metadata);
+    }
+
+    private StringBuilder addJvmParamsToConfigurationInPluginText(StringBuilder plugin) {
+        Map<String, String> jvmParams = new TreeMap<String, String>(pluginInfo.getSelightsJvmParams());
+        return addMapConfigurationToPluginText(plugin,SEALIGHTS_JVM_PARAMS_KEY , jvmParams);
+    }
+
+    private StringBuilder addBuildScannerParamsToConfigurationInPluginText(StringBuilder plugin) {
+        Map<String, String> jvmParams = new TreeMap<String, String>(pluginInfo.getBuildScannerParams());
```

Comment:
Why do you need those 3 functions? Seems like you create call ```addMapConfigurationToPluginText``` directly. The only difference is the key and map instance.